### PR TITLE
3115 better search pages error

### PR DESCRIPTION
**Pull request by kaguillera**
Please review for #3115

### CODE REVIEW

File: tests/py/test_pages.py [hunk 1]
```diff
@@ -166,3 +166,26 @@ def test_new_participant_can_edit_profile(self):
         self.make_participant('alice', claimed_time='now')
         body = self.client.GET("/alice/", auth_as="alice").body
         assert b'Edit' in body
+
+    def test_search_results_error_pages_twitter(self):
+<<<<<<< Updated upstream
```
Comment (whit537) [inline tests/py/test_pages.py]: You've got a merge conflict here (the `<<<<<<<`). Please resolve.

**Comment by Changaco** [comment]
I don't think messing with `error.spt` is the right approach, I think should modify the `get_account_elsewhere()` function instead.

**Comment by kaguillera** [comment]
I removed the code from error.spt and rewrote a more reasonable error statement in the `check_api_response_status()` method since that is where the Response is raised first when a user does not exist on a platform. I guess it can also be done in the `get_account_elsewhere()` as you suggested @Changaco but that `Response(404)' is only raised when the no platform is given

**Comment by Changaco** [comment]
`check_api_response_status()` is a generic method, it's not limited to user lookups, changing the error message there is incorrect.

**Comment by kaguillera** [comment]
I  noticed that @Changaco but that is the place where the 404 response from the platform APIs request is being caught and the raising of the aspen Response. I will look around again an see if I can catch it before. if you have any other suggestion I am open to it. As I said before

**Comment by kaguillera** [comment]
Did not finish... `get_account_elsewhere()` is not reached because `check_api_response_status()` is called first in the case of the search.

**Comment by kaguillera** [comment]
I finally got to capture and raise the 404 Response in `get_account_elsewhere()'  @Changaco @whit537 could you review and tell me what you think.

**Comment by kaguillera** [comment]
Travis keeps failing can anyone tell me why that is?

**Comment by Changaco** [comment]
This is still not correct: you're turning all responses into 404s, even the ones that aren't actually 404s.

**Comment by kaguillera** [comment]
Thanks did not realize that. Will fix it.

**Comment by kaguillera** [comment]
I figured out the function in the test_elsewhere.py test script that seems to be affecting the test function in test_pages.py that I have been working on. It is the test_user_pages(self): function. My "educated guess” is that after that function does a call to get_user_info it is not successfully cleared and thus retaining the user information in session/cache or something. I was wondering is you could point me in the right direction of if my guess is right and if so who can I clear/reset thing so that I have a clean call to the platform API to get the proper response.

**Comment by whit537** [comment]
> My "educated guess” is that after that function does a call to get_user_info it is not successfully cleared and thus retaining the user information in session/cache or something.

@kaguillera In this situation I find it helpful to throw a `set_trace` in the test so I can inspect the database and in-memory caches to find the bit of info that's not being cleared. Off the top of my head I don't have a strong sense of what it might be that's not being cleared. I think we still have some in-memory structures related to signing in via elsewhere account? @Changaco @rohitpaulk @clone1018 @seanlinsley Do you remember where these are?

**Comment by Changaco** [comment]
Ready for review.

@kaguillera The fix you were looking for is in 4f69b5b470290ac7bde1bd8e98ee70e723196d57.

**Comment by kaguillera** [comment]
@Changaco Please review, is this failure because of something I did?

**Comment by Changaco** [comment]
Rebased on master.

@kaguillera Yes, the merge you did wasn't good.

**Comment by Changaco** [comment]
Rebased on master again.

**Comment by kaguillera** [comment]
Yep looks good, well done Changaco.

**Comment by Changaco** [comment]
I've added a couple more commits for i18n.

**Comment by whit537** [comment]
@kaguillera Since @Changaco has implicitly signed off on your commits, then if you are okay with his commits then it is your prerogative to merge.

**Pull request merged**